src/client.rs

```diff
@@ -109,10 +104,9 @@ impl Client {
     /// # });
     /// ```
     pub async fn get_index(&self, uid: impl AsRef<str>) -> Result<Index, Error> {
-        match self.get_raw_index(uid).await {
-            Ok(raw_idx) => Ok(raw_idx.into_index(self)),
-            Err(error) => Err(error),
-        }
+        let mut idx = self.index(uid.as_ref());
+        idx.fetch_info().await?;
+        Ok(idx)
```

Comment:
We cannot just return: `idx.fetch_info().await?` ?

Reply:
No, `fetch_info` returns `Result<(), Error>`, as it fetches the information and updates self, and this method, `Client::get_index` returns `Result<Index, Error>`.